pytoda/preprocessing/crawlers.py

```diff
@@ -124,6 +128,59 @@ def get_smiles_from_pubchem(
             return smiles
         except urllib_error.HTTPError:
             if option == 'CanonicalSMILES':
-                logger.warninig(f'Did not find any result for drug: {drug}')
-                return []
+                logger.warning(f'Did not find any result for drug: {drug}')
+                return ''
             continue
+
+
+def remove_pubchem_smiles(smiles_list: Iterable[str]) -> List:
+    """
+    Function for removing PubChem molecules from an iterable of smiles.
+    Args:
+        smiles_list (Iterable[str]): many SMILES strings.
+    Returns:
+        List[str]:  Filtered list of SMILES, all SMILES pointing to PubChem
+            molecules are removed.
+    """
+
+    if not isinstance(smiles_list, Iterable):
+        raise TypeError(f'Please pass Iterable, not {type(smiles_list)}')
+
+    canonicalizer = Canonicalization(sanitize=False)
+    filtered = filterfalse(lambda x: is_pubchem(x), smiles_list)
+    # Canonicalize molecules and filter again (sanity check)
+    filtered_canonical = filterfalse(
+        lambda x: is_pubchem(canonicalizer(x)), filtered
+    )
+    return list(filtered_canonical)
+
+
+def query_pubchem(smiles: str) -> Tuple[bool, int]:
+    """
+    Queries pubchem for a given SMILES.
+    Args:
+        smiles (str): A SMILES string.
+    Returns:
+        Tuple[bool, int]
+            bool: Whether or not SMILES is known to PubChem.
+            int: PubChem ID of matched SMILES, -1 if SMILES was not found.
+    """
+    if not isinstance(smiles, str):
+        raise TypeError(f'Please pass str, not {type(smiles)}')
+    try:
+        result = pcp.get_compounds(smiles, 'smiles')[0]
+    except BadRequestError:
+        logger.warning(f'Skipping SMILES. BadRequestError with: {smiles}')
+        return (False, -1)
```

Comment:
```suggestion
        return (False, -2)
```
Just an idea to be able to distinguish the types of "not found"